tests/integration/veth_test.py

```diff
@@ -316,3 +317,18 @@ def veth_interface(ifname, peer):
             }
         )
         libnmstate.apply(d_state)
+
+
+@pytest.mark.skipif(
+    nm_major_minor_version() <= 1.28,
+    reason="Modifying veth interfaces is not supported on NetworkManager.",
+)
+@pytest.mark.xfail(reason="https://bugzilla.redhat.com/1943525", strict=True)
+def test_remove_peer_connection():
+    with veth_interface(VETH1, VETH1PEER) as desired_state:
+        desired_state[Interface.KEY][0][Interface.STATE] = InterfaceState.ABSENT
+        libnmstate.apply(desired_state)
+        assert (
+            cmdlib.exec_cmd(f"nmcli connection show {VETH1PEER}".split())[0]
```

Comment:
If you are using `nmcli` to check whether `VETH1PEER` still exists, you should place this test in to `tests/integration/nm` folder.

You may also change to `assert len(statelib.show_only((VETH1PEER, ))[Interface.NAME]) == 0` which does not require the nm specific command.

Reply:
After taking `VETH1` absent, this veth pair was removed from kernel, including `VETH1PEER`, so `len(statelib.show_only((VETH1PEER, ))[Interface.NAME])` is 0, there's no problem here.
But the NM profile of `VETH1PEER` remained, that's what I need to check(expecting the NM profile was deleted too)